src/UNL/UCBCN/Manager/CreateCalendar.php

```diff
@@ -70,6 +98,22 @@ private function createCalendar($post_data)
         $calendar = new Calendar;
         $calendar->account_id = $account->id;
         $calendar->name = $post_data['name'];
+
+        # name and shortname are required
+        if (empty($post_data['name']) || empty($post_data['shortname'])) {
+            throw new ValidationException('Calendar name and shortname are required.');
+        }
+
+        # check that the shortname will match the regex it needs to
+        if (!preg_match('/^[a-zA-Z-_0-9]+$/', $post_data['shortname'])) {
```

Comment:
ideally, the same variable would handle both the regex for the router and validation, that way if you update one, both are updated.